### PR TITLE
DSG-623/Set-spring-profile-dynamically

### PR DESCRIPTION
**Pull request by afcalderonm**
## Description

configure spring profile to be set dynamically

[DSG-623](https://dol-ewf.atlassian.net/browse/DSG-623)

## Motivation and Context

We recently created the dsgov-logging lib which has the configuration files:

logback-local.xml

logback-cloud.xml

From this it was decided that we will only use 2 spring profiles, local and cloud, but this affected some apps for cloud deployment, more specifically in dev env.

## How Has This Been Tested?

tested locally and saw that the active profile was local after the adjustment, it was default before.

## Types of changes

Patch

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
  Minor
- [ ] New feature (non-breaking change which adds functionality)
  Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met.

[DSG-623]: https://dol-ewf.atlassian.net/browse/DSG-623?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: skaffold.yaml
Comment (jgnuvalence) [inline skaffold.yaml:1]: please add:

`googleCloudBuild:`
`        projectId: dsgov-services`
`        gradleImage: gradle:7.6-jdk11-alpine`

in profiles.build

[the remark this answers is not in the document]
Comment (jgnuvalence) [reply skaffold.yaml:1]: after doing this we have to change the profiles.deploy.setValueTemplates with:

`image.repository: "{{.IMAGE_REPO_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_work_manager}}"`
            
`image.tag: "{{.IMAGE_TAG_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_work_manager}}@{{.IMAGE_DIGEST_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_work_manager}}"`

**Comment by jgnuvalence** [comment]
I leave the comments of the things I see that are missing, but I would really like @dtsong  to review as well

**Comment by jgnuvalence** [comment]
@afcalderonm please also see the changes done in this [PR](https://github.com/Nuvalence/dsgov-work-manager/pull/71/files) for `.ci/cloudbuild-push-request.yaml`
and for `service/build.gradle` note that the jib configurations was removed, you need to do the same, after that you need to check that the spring profile es set correctly in minikube when you deploy

**Review by jgnuvalence — APPROVED** [review]
LGTM

File: skaffold.yaml [hunk 1]
```diff
@@ -5,20 +5,25 @@ metadata:
...
       helm:
         releases:
         - name: dsgov-audit-service
           chartPath: helm/audit-service
           # Ref: https://skaffold.dev/docs/deployers/helm/#image-configuration
           setValueTemplates:
-            image.repository: "{{.IMAGE_REPO_dsgov_audit_service}}"
-            image.tag: "{{.IMAGE_TAG_dsgov_audit_service}}@{{.IMAGE_DIGEST_dsgov_audit_service}}"
+            image.repository: "{{.IMAGE_REPO_us_east4_docker_pkg_dev_dsgov_services_application_dsgov-audit-service}}"
+            image.tag: "{{.IMAGE_TAG_us_east4_docker_pkg_dev_dsgov_services_application_dsgov-audit-service}}@{{.IMAGE_DIGEST_us_east4_docker_pkg_dev_dsgov_services_application_dsgov-audit-service}}"
```
Comment (dtsong) [inline skaffold.yaml:26]: We will want all of these to have underscores like so:

```
IMAGE_REPO_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_audit_service
IMAGE_TAG_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_audit_service
IMAGE_DIGEST_us_east4_docker_pkg_dev_dsgov_services_application_dsgov_audit_service
```

**Review by dtsong — CHANGES_REQUESTED** [review]
One comment, otherwise looks good!

**Review by dtsong — APPROVED** [review]

**Pull request merged**